### PR TITLE
adding capabitlity to add new peer without opening channel. able to delete the peer as well providing a channel theres no active channel

### PR DESCRIPTION
**Pull request by eoliveros**
- added html pages:
    - peer_management.html
    - peer_connector.html
- added the ln_wallet_endpoints for the above pages.
- added the links in lightning/index.html

### CODE REVIEW

**Review by djpnewton — CHANGES_REQUESTED** [review]
merge the peer connect form into the peer management page

File: src/ln_wallet_endpoint.py [hunk 1]
```diff
@@ -199,6 +199,30 @@ def channel_opener():
             flash(Markup(e.args[0]), 'danger')
     return render_template('lightning/channel_opener.html', funds_dict=LnRpc().list_funds())
 
+@ln_wallet.route('/peer_management', methods=['GET', 'POST'])
+@roles_accepted(Role.ROLE_ADMIN)
+def peer_management():
+    rpc = LnRpc()
+    if request.method == 'POST':
+        if request.form['form-name'] == 'close_peer_form':
```
Comment (djpnewton) [inline src/ln_wallet_endpoint.py:207]:
```suggestion
        if request.form['form-name'] == 'peer_close_form':
```
make similar to other form name ('peer_connect_form')

File: src/ln_wallet_endpoint.py [hunk 2]
```diff
@@ -199,6 +199,30 @@ def channel_opener():
...
+            peer_id = request.form["peerId"]
```
Comment (djpnewton) [inline src/ln_wallet_endpoint.py:208]:
```suggestion
            peer_id = request.form['peerId']
```
keep quotes consistent

File: src/ln_wallet_endpoint.py [hunk 3]
```diff
@@ -199,6 +199,30 @@ def channel_opener():
...
+                LnRpc().disconnect_peer(peer_id)
+                flash(Markup(f'successfully disconnected id: {peer_id}'), 'success')
+            except Exception as e:
+                flash(Markup(f'{e.error["message"]}'), 'danger') # pyright: ignore [reportGeneralTypeIssues]
+        elif request.form['form-name'] == 'peer_connect_form':
+            peer_id = request.form["peerId"]
+            try:
+                rpc = LnRpc()
+                rpc.connect_node(peer_id)
+                node_id = peer_id.split("@")
```
Comment (djpnewton) [inline src/ln_wallet_endpoint.py:219]:
```suggestion
                node_id = peer_id.split('@')
```

File: src/ln_wallet_endpoint.py [hunk 4]
```diff
@@ -199,6 +199,30 @@ def channel_opener():
...
+def peer_management():
+    rpc = LnRpc()
+    if request.method == 'POST':
+        if request.form['form-name'] == 'close_peer_form':
+            peer_id = request.form["peerId"]
+            try:
+                LnRpc().disconnect_peer(peer_id)
+                flash(Markup(f'successfully disconnected id: {peer_id}'), 'success')
+            except Exception as e:
+                flash(Markup(f'{e.error["message"]}'), 'danger') # pyright: ignore [reportGeneralTypeIssues]
```
Comment (djpnewton) [inline src/ln_wallet_endpoint.py:213]:
```suggestion
                flash(Markup(f'{e}'), 'danger')
```

File: src/ln_wallet_endpoint.py [hunk 5]
```diff
@@ -199,6 +199,30 @@ def channel_opener():
...
+        elif request.form['form-name'] == 'peer_connect_form':
+            peer_id = request.form["peerId"]
+            try:
+                rpc = LnRpc()
+                rpc.connect_node(peer_id)
+                node_id = peer_id.split("@")
+                flash(Markup(f'successfully connected peer: {node_id[0]}'), 'success')
+            except Exception as e:
+                flash(Markup(f'{e.error["message"]}'), 'danger') # pyright: ignore [reportGeneralTypeIssues]
+    peers = rpc.list_peers()["peers"]
```
Comment (djpnewton) [inline src/ln_wallet_endpoint.py:223]:
```suggestion
                flash(Markup(f'{e}'), 'danger')
    peers = rpc.list_peers()['peers']
```

File: src/models.py [hunk 6]
```diff
@@ -71,6 +71,7 @@ def total_of_asset(cls, session: Session, user: User, asset: str, l2_network: st
 
 roles_users = Table(
     'roles_users',
+    BaseModel.metadata,
```
Comment (djpnewton) [inline src/models.py:74]: this shouldnt be here, also will cause merge error

File: src/models.py [hunk 7]
```diff
@@ -202,6 +203,7 @@ def __str__(self):
 
 permissions_api_keys = Table(
     'permissions_api_keys',
+    BaseModel.metadata,
```
Comment (djpnewton) [inline src/models.py:206]: this shouldnt be here, also will cause merge error

File: src/templates/lightning/peer_management.html [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+                    {% endif %}
+                    <div class="collapse" id="collapseExample{{ loop.index }}">
+                        <div class="card card-body">
+                            <ul class="list-unstyled my-3">
+                                <li>
+                                    <strong>ID:</strong> {{ peer.id }}
+                                </li>
+                                <li>
+                                    <strong>Connected:</strong> {{ peer.connected }}
+                                </li>
```
Comment (djpnewton) [inline src/templates/lightning/peer_management.html:64]:
```suggestion                        
                      <button type="button"
                              class="btn {% if not peer.channels %}btn-light{% else %}btn-warning{% endif %} btn-lg btn-block"
                              data-toggle="collapse"
                              href="#collapseExample{{ loop.index }}"
                             role="button"
                             aria-expanded="false"
                              aria-controls="collapseExample">
                          <header class="d-flex justify-content-between mb-2">
                              id: {{ peer.id }}
                          </header>
                      </button>
                    {% endif %}
                    <div class="collapse" id="collapseExample{{ loop.index }}">
                        <div class="card card-body">
                            <ul class="list-unstyled my-3">
                                <li>
                                    <strong>ID:</strong> {{ peer.id }}
                                </li>
                                <li>
                                    <strong>Connected:</strong> {{ peer.connected }}
                                </li>
                                {% if peer.channels %}
                                <li>
                                    <strong>Has channel:</strong> true
                                </li>
                                {% endif %}
```

**Review by djpnewton — CHANGES_REQUESTED** [review]
seems to work, lightningd wont let you disconnect from a peer with channels

File: src/ln_wallet_endpoint.py (hunk 1 above)
Comment (eoliveros) [reply src/ln_wallet_endpoint.py]: changed per above.

File: src/ln_wallet_endpoint.py (hunk 2 above)
Comment (eoliveros) [reply src/ln_wallet_endpoint.py]: changed per above.

File: src/ln_wallet_endpoint.py (hunk 3 above)
Comment (eoliveros) [reply src/ln_wallet_endpoint.py]: changed per above.

File: src/ln_wallet_endpoint.py (hunk 4 above)
Comment (eoliveros) [reply src/ln_wallet_endpoint.py]: changed per above.

File: src/ln_wallet_endpoint.py (hunk 5 above)
Comment (eoliveros) [reply src/ln_wallet_endpoint.py]: changed per above.

File: src/models.py (hunk 7 above)
Comment (eoliveros) [reply src/models.py]: changed per above.

File: src/templates/lightning/peer_management.html (hunk 8 above)
Comment (eoliveros) [reply src/templates/lightning/peer_management.html]: changed with a few differences

File: src/models.py (hunk 6 above)
Comment (eoliveros) [reply src/models.py]: removed the line above as its been implemented on another change.

**Review by djpnewton — CHANGES_REQUESTED** [review]
looks good, can you squash the commits into a single commit

**Review by djpnewton — APPROVED** [review]

**Pull request merged**